src/db-connection.ts

```diff
@@ -0,0 +1,79 @@
+import { config } from 'node-config-ts'
+import type { Knex } from 'knex'
+import knex from 'knex'
+import snakeCase from 'lodash.snakecase'
+import camelCase from 'lodash.camelcase'
+import { logger } from './logger/index.js'
+
+const KNEX_TABLES = ['knex_migrations', 'knex_migrations_lock']
+
+function toCamelCase(value) {
+  if (Array.isArray(value)) {
+    return value.map(toCamelCase)
+  }
+
+  if (typeof value === 'object') {
+    return Object.fromEntries(Object.entries(value).map(([key, value]) => [camelCase(key), value]))
+  }
+
+  if (typeof value === 'string') {
+    return camelCase(value)
+  }
+
+  return value
+}
+
+async function runMigrations(connection: Knex) {
+  const [_, pendingMigrations] = await connection.migrate.list()
+  if (pendingMigrations.length > 0) {
+    logger.imp(`Running ${pendingMigrations.length} migrations`)
+    await connection.migrate.latest()
+  }
+}
+
+export async function createDbConnection() {
+  const knexConfig: Knex.Config = {
+    ...config.db,
+    // transforms identifier names from camel case to snake case. Prevents useage of snakecase in code to maintain consistency.
+    wrapIdentifier: (value, origWrap): string => origWrap(snakeCase(value)),
+    // modifies returned rows transforming identifiers from snake case to camel case. Prevents having to do it manually in code.
+    postProcessResponse: (result) => toCamelCase(result),
+  }
+
+  let connection
+  try {
+    connection = knex(knexConfig)
+  } catch (e) {
+    throw new Error(`Database connection failed: ${e}`)
+  }
+
+  await runMigrations(connection).catch((err) => {
+    throw new Error(`Migrations have failed: ${err}`)
+  })
+
+  return connection
+}
+
+/**
+ * USED FOR TESTING
+ * Clears all tables
+ * @param connection
+ */
+export async function clearTables(connection: Knex): Promise<void> {
```

Comment:
@stephhuynh18 Have you checked against Postgres? There are few slight surprising differences between SQLite and PG backends for Knex. One of them is `count` response, another is about table names, that I remember.

I am pretty sure you have, but it's worth stating that explicitly. I am just worried about some bug to appear out of the blue on Pg.

Reply:
UPD: Please, disregard this.